lib/cdo/config.rb

```diff
@@ -8,11 +8,15 @@ class Config < OpenStruct
     # Soft-freeze: Don't allow any config items to be created/modified,
     # but allow stubbing for unit tests.
     def freeze
-      @table.each_key(&method(:new_ostruct_member!))
+      # Some implementation changes in OpenStruct in Ruby 3.0 affect behavior
+      # that this class is relying on, so we need to include some special logic
+      # here. See https://bugs.ruby-lang.org/issues/15409#note-9
+      @table.each_key {|k| new_ostruct_member!(k.to_sym)}
       @frozen = true
     end
 
     def method_missing(key, *args)
+      return self[key.to_sym] if args.empty? && defined?(key.to_sym) # accommodate https://bugs.ruby-lang.org/issues/15409#note-9
```

Comment:
@wjordan is `defined?` here supposed to be checking to see if `key.to_sym` is one of the values that's been set on this struct? Because I don't think it's doing that. Should this be something like:

```suggestion
      return self[key.to_sym] if args.empty? && @table.key?(key.to_sym) # accommodate https://bugs.ruby-lang.org/issues/15409#note-9
```

Reply:
I think I had meant to check if an accessor method was already defined (via `new_ostruct_member!`) on the symbolized key, in which case the two should be similar. I'm not sure though, this code was just a quick workaround for issues in the updated ostruct - probably worth testing it out locally and using whatever works.

Longer-term, to clean this up I think the CDO config should be rewritten to just define the convenience accessor-methods it wants directly on itself, and not bother inheriting OpenStruct at all (as [recommended](https://github.com/ruby/ostruct/blob/c535a406f8e56b0d2d4fd470c65ec070f316b190/lib/ostruct.rb#L107) by the gem).

Reply:
That makes sense! It looks like we're getting a false positive with this approach, specifically on https://github.com/code-dot-org/code-dot-org/blob/d6e643accbef7ab0b3684b7a5092c28faa1bc587/lib/test/cdo/test_config.rb#L64

I'll move forward with `@table.key?` for now, and created https://codedotorg.atlassian.net/browse/INF-522 to track the longer-term work.

Thank you!